### PR TITLE
Explicitly deleting default connection on js side

### PR DESCRIPTION
**Pull request by hannes**

### CODE REVIEW

**Comment by hannes** [comment]
Hopefully fixes #4164, @zenazn @danielchiu could you guys check if this fixes your issue? I find the typescript example challenging to run here, my apologies

File: tools/nodejs/lib/duckdb.js [hunk 1]
```diff
@@ -305,16 +310,20 @@ var default_connection = function (o) {
...
 /**
+ * Closes database instance
  * @method
  * @param callback
  * @return {void}
  */
-Database.prototype.close;
+Database.prototype.close = function() {
+    this.default_connection = null
+    this.close_internal();
```
Comment (danielchiu) [inline tools/nodejs/lib/duckdb.js:320]: I believe this needs to call the input callback, so the caller knows when the close is done (same API that other functions use).

Specifically, I was able to get this to work using
```suggestion
     this.close_internal(arguments['0']);
```

**Review by danielchiu — COMMENTED** [review]
Hi @hannes, thank you!

It almost worked; I added a comment for the fix I needed. After the change, running the script `duckdbRepeatedlyCreateTables.ts` works smoothly on my machine.

File: tools/nodejs/lib/duckdb.js (hunk 1 above)
Comment (hannes) [reply tools/nodejs/lib/duckdb.js]: Indeed that was an oversight. I added your suggestion (in a slightly different way).

Comment (danielchiu) [reply tools/nodejs/lib/duckdb.js]: lgtm!

File: tools/nodejs/lib/duckdb.js [hunk 2]
```diff
@@ -305,16 +310,20 @@ var default_connection = function (o) {
...
 /**
+ * Closes database instance
  * @method
  * @param callback
  * @return {void}
  */
-Database.prototype.close;
+Database.prototype.close = function() {
+    this.default_connection = null
+    this.close_internal.apply(arguments);
```
Comment (Mause) [inline tools/nodejs/lib/duckdb.js:320]:
```suggestion
    this.close_internal.apply(this, arguments);
```

**Review by Mause — CHANGES_REQUESTED** [review]

**Pull request merged**